vivisect/analysis/aarch64/emulation.py

```diff
@@ -0,0 +1,486 @@
+import logging
+
+import visgraph.pathcore as vg_path
+
+import vivisect
+import vivisect.exc as v_exc
+import vivisect.impemu.monitor as viv_monitor
+import vivisect.analysis.generic.codeblocks as viv_cb
+
+import envi
+import envi.archs.aarch64 as e_a64
+
+from envi.archs.aarch64.regs import *
+from envi.archs.aarch64.const import *
+
+from vivisect.const import *
+
+logger = logging.getLogger(__name__)
+
+
+class AnalysisMonitor(viv_monitor.AnalysisMonitor):
+
+    def __init__(self, vw, fva, verbose=True):
+        viv_monitor.AnalysisMonitor.__init__(self, vw, fva)
+        self.retbytes = None
+        self.badops = vw.arch.archGetBadOps()
+        self.last_lr_pc = 0
+        self.strictops = False
+        self.returns = False
+        self.infloops = []
+        self.switchcases = 0
+
+    def prehook(self, emu, op, starteip):
```

Comment:
So a ton of this file seems to be scrimmed from the ARM emulator pass (almost verbatim). Is there anyway we can combine the two?